pyerrors/input/json.py

```diff
@@ -0,0 +1,328 @@
+        elif isinstance(io, list):
+            d['obsdata'].append(write_List_to_dict(io))
+        elif isinstance(io, np.ndarray):
+            d['obsdata'].append(write_Array_to_dict(io))
+
+    jsonstring = json.dumps(d, indent=indent, cls=my_encoder, ensure_ascii=False)
+    # workaround for un-indentation of delta lists
+    jsonstring = jsonstring.replace('    "[', '    [').replace(']",', '],').replace(']"\n', ']\n')
+
+    return jsonstring
+
+
+def dump_to_json(ol, fname, description='', indent=1, gz=True):
+    """Export a list of Obs or structures containing Obs to a .json(.gz) file
+
+    Parameters
+    -----------------
+    ol : list
+        List of objects that will be exported. At the moments, these objects can be
+        either of: Obs, list, np.ndarray
+        All Obs inside a structure have to be defined on the same set of configurations.
+    fname : str
+        Filename of the output file
+    description : str
+        Optional string that describes the contents of the json file
+    indent : int
+        Specify the indentation level of the json file. None or 0 is permissible and
+        saves disk space.
+    gz : bool
+        If True, the output is a gzipped json. If False, the output is a json file.
+    """
+
+    jsonstring = create_json_string(ol, fname, description, indent)
+
+    if not fname.endswith('.json') and not fname.endswith('.gz'):
+        fname += '.json'
+
+    if gz:
+        if not fname.endswith('.gz'):
+            fname += '.gz'
+
+        fp = gzip.open(fname, 'wb')
+        fp.write(jsonstring.encode('utf-8'))
+    else:
+        fp = open(fname, 'w', encoding='utf-8')
+        fp.write(jsonstring)
+    fp.close()
+
+
+def load_json(fname, verbose=True, gz=True, full_output=False):
+    """Import a list of Obs or structures containing Obs from a .json.gz file.
+    The following structures are supported: Obs, list, np.ndarray
+    If the list contains only one element, it is unpacked from the list.
+
+    Parameters
+    -----------------
+    fname : str
+        Filename of the input file
+    verbose : bool
+        Print additional information that was written to the file.
+    gz : bool
+        If True, assumes that data is gzipped. If False, assumes JSON file.
+    full_output : bool
+        If True, a dict containing auxiliary information and the data is returned.
+        If False, only the data is returned.
+    """
+
+    def _gen_obsd_from_datad(d):
+        retd = {}
+        retd['names'] = []
+        retd['idl'] = []
+        retd['deltas'] = []
+        retd['is_merged'] = {}
+        for ens in d:
+            for rep in ens['replica']:
+                retd['names'].append(rep['name'])
+                retd['idl'].append([di[0] for di in rep['deltas']])
+                retd['deltas'].append(np.array([di[1:] for di in rep['deltas']]))
+                retd['is_merged'][rep['name']] = rep.get('is_merged', False)
+        retd['deltas'] = np.array(retd['deltas'])
```

Comment:
The visible deprecation warning stems from this line as it can happen that the individual entries of the list do not have the same length. Is it necessary to cast to an `numpy.ndarray` here?

Reply:
I don't see the warning, but this may be related to the python version. It is convenient to use the array, because I can access columns, but it should be possible to restructure the code, such that it can be removed.

Reply:
The warning appears from numpy 1.19 onward. And could stop the code from running in the future so I would suggest a workaround.

Reply:
I think just removing the array cast does not break the code. The iterations you are using to reconstruct the objects should also work with standard lists.

Reply:
Turns out that the newly inserted cast in line 228 is enough. I deleted line 230, as it is not necessary anymore. Does this resolve the issue?

Reply:
Yes that works!